Modules/Core/Common/include/itkNeighborhoodAllocator.h

```diff
@@ -92,6 +104,22 @@ class NeighborhoodAllocator
     return *this;
   }
 
+
+  /** Move-assignment. */
+  Self& operator=(Self&& other) ITK_NOEXCEPT
+  {
+    if (this != &other)
+    {
+      this->Deallocate();
```

Comment:
Nice. I guess because `other.m_Data` is `nullptr`, the call of Deallocate (`delete[]`) at the destruction of `other` is safe (it won't delete the address).

Reply:
@phcerdan Thanks! Indeed, `other.m_Data` should be set to `nullptr`, to allow `other` to be destructed safely.

The move-assignment of my initial review-push had a memory leak, as it initially did not yet do `this->Deallocate()`. Fortunately I spotted the bug in time.  `:-)`